File: ios/runanywhere-sdks-latest/sdk/runanywhere-flutter/packages/runanywhere/lib/foundation/security/secure_storage_keys.dart
/// SecureStorageKeys
///
/// Keychain/secure storage key constants
///
/// Reference: sdk/runanywhere-swift/Sources/RunAnywhere/Foundation/Security/KeychainManager.swift
/// Reference: sdk/runanywhere-react-native/packages/core/src/Foundation/Security/SecureStorageKeys.ts
///
/// These keys are used for:
/// - iOS: Keychain (survives app reinstalls)
/// - Android: EncryptedSharedPreferences (survives app reinstalls)
class SecureStorageKeys {
  SecureStorageKeys._(); // Prevent instantiation

  // SDK Core
  static const apiKey = 'com.runanywhere.sdk.apiKey';
  static const baseURL = 'com.runanywhere.sdk.baseURL';
  static const environment = 'com.runanywhere.sdk.environment';

  // Device Identity (survives app reinstalls)
  static const deviceUUID = 'com.runanywhere.sdk.device.uuid';
  static const deviceRegistered = 'com.runanywhere.sdk.device.isRegistered';

  // Authentication Tokens
  static const accessToken = 'com.runanywhere.sdk.accessToken';
  static const refreshToken = 'com.runanywhere.sdk.refreshToken';
  static const tokenExpiresAt = 'com.runanywhere.sdk.tokenExpiresAt';

  // User/Org Identity
  static const deviceId = 'com.runanywhere.sdk.deviceId';
  static const userId = 'com.runanywhere.sdk.userId';
  static const organizationId = 'com.runanywhere.sdk.organizationId';
}

File: ios/runanywhere-sdks-latest/sdk/runanywhere-react-native/packages/core/src/Foundation/Security/SecureStorageKeys.ts
/**
 * SecureStorageKeys.ts
 *
 * Keychain/secure storage key constants
 *
 * Reference: sdk/runanywhere-swift/Sources/RunAnywhere/Foundation/Security/KeychainManager.swift
 */

/**
 * Keys for secure storage (keychain on iOS, keystore on Android)
 *
 * These match the iOS KeychainKey enum values exactly.
 */
export const SecureStorageKeys = {
  // SDK Core
  apiKey: 'com.runanywhere.sdk.apiKey',
  baseURL: 'com.runanywhere.sdk.baseURL',
  environment: 'com.runanywhere.sdk.environment',

  // Device Identity
  deviceUUID: 'com.runanywhere.sdk.device.uuid',

  // Authentication Tokens
  accessToken: 'com.runanywhere.sdk.accessToken',
  refreshToken: 'com.runanywhere.sdk.refreshToken',
  tokenExpiresAt: 'com.runanywhere.sdk.tokenExpiresAt',

  // User/Org Identity
  deviceId: 'com.runanywhere.sdk.deviceId',
  userId: 'com.runanywhere.sdk.userId',
  organizationId: 'com.runanywhere.sdk.organizationId',
} as const;

export type SecureStorageKey =
  (typeof SecureStorageKeys)[keyof typeof SecureStorageKeys];
